### PR TITLE
Add ISPC language support

### PR DESCRIPTION
**Pull request by fab4100**
- Add ISPC parser config
- Add ISPC queries

The two commits add grammar and queries for the Intel implicit SPMD program
compiler ([ISPC](https://ispc.github.io/ispc.html)).

### CODE REVIEW

File: queries/ispc/highlights.scm [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+[
+  "cif"
+] @conditional
+
+[
+  "varying"
+  "uniform"
+] @type.qualifier
+
+"__regcall" @attribute
```
Comment (amaanq) [inline queries/ispc/highlights.scm:38]:
```suggestion
"__regcall" @storageclass
```

File: queries/ispc/injections.scm [hunk 2]
```diff
@@ -0,0 +1,3 @@
+(preproc_arg) @c
+
+(comment) @comment
```
Comment (amaanq) [inline queries/ispc/injections.scm:3]:
```suggestion
; inherits: c
```

**Review by amaanq — CHANGES_REQUESTED** [review]

**Comment by amaanq** [comment]
Mostly good, left a couple notes

File: queries/ispc/highlights.scm (hunk 1 above)
Comment (fab4100) [reply queries/ispc/highlights.scm:38]: The `__regcall` attribute does not change the storage class of an object, it affects function argument passing in registers (https://www.intel.com/content/www/us/en/docs/cpp-compiler/developer-guide-reference/2021-9/c-c-calling-conventions.html).  A similar attribute is `__vectorcall` which is already handled in `c/highlights.scm` as `@attribute`.  I would treat `__regcall` the same as `__vectorcall`.

Comment (amaanq) [reply queries/ispc/highlights.scm:38]: I was thinking of storageclass in a loose sense that a type of calling convention can be linked to its layout/storage at runtime, but I didn't see other similar ones already marked as attribute...seems fine then

**Review by amaanq — APPROVED** [review]

File: queries/ispc/highlights.scm (hunk 1 above)
Comment (fab4100) [reply queries/ispc/highlights.scm:38]: Since Intel C/C++ compilers support `__regcall` it should probably be moved into `c/highlights.scm` anyway.

**Pull request merged**